src/components/atoms/textarea.tsx

```diff
@@ -1,23 +1,104 @@
-import React, { forwardRef } from "react";
+import React, { ChangeEvent, FocusEvent, forwardRef, useCallback, useEffect, useId, useState } from "react";
 
 import { TextArea } from "@interfaces/components";
 import { cn } from "@utilities";
 
-export const Textarea = forwardRef<HTMLTextAreaElement, Partial<TextArea>>((props, ref) => {
-	const { className, disabled, isError, placeholder = "Enter text", ...rest } = props;
-
-	const baseStyle = cn(
-		"border border-gray-950 bg-black text-base text-white",
-		"w-full pb-2 pl-4 pr-1.5 pt-3.5",
-		"placeholder:font-light placeholder:text-white hover:placeholder:font-medium",
-		"scrollbar rounded-lg transition hover:border-white focus:border-white",
-		{ "border-error": isError },
-		{ "pointer-events-none select-none": disabled },
-		{ "placeholder:text-gray-500": disabled },
+export const Textarea = forwardRef<HTMLTextAreaElement, TextArea>((props, ref) => {
+	const {
+		className,
+		defaultValue = "",
+		disabled = false,
+		isError = false,
+		isRequired = false,
+		label,
+		onBlur,
+		onChange,
+		placeholder,
+		value,
+		...rest
+	} = props;
+
+	const [isFocused, setIsFocused] = useState(false);
+	const [textareaValue, setTextareaValue] = useState(value ?? defaultValue);
+	const [hasValue, setHasValue] = useState(() => !!textareaValue);
+
+	useEffect(() => {
+		if (value !== undefined) {
+			setTextareaValue(value);
+			setHasValue(!!value);
+		}
+	}, [value]);
+
+	const handleFocus = useCallback(() => {
+		setIsFocused(true);
+	}, []);
+
+	const handleBlur = useCallback(
+		(event: FocusEvent<HTMLTextAreaElement>) => {
+			setIsFocused(false);
+			setHasValue(!!textareaValue);
+			onBlur?.(event);
+		},
+		[onBlur, textareaValue]
+	);
+
+	const handleChange = useCallback(
+		(event: ChangeEvent<HTMLTextAreaElement>) => {
+			const newValue = event.target.value;
+			setTextareaValue(newValue);
+			setHasValue(!!newValue);
+			onChange?.(event);
+		},
+		[onChange]
+	);
```

Comment:
What if we do like this?
It would be easier to read and to work with it in the future?

```suggestion
	const [state, setState] = React.useState({
		isFocused: false,
		textareaValue: value ?? defaultValue,
		hasValue: !!(value ?? defaultValue),
	});

	useEffect(() => {
		if (value !== undefined) {
			setState((prev) => ({
				...prev,
				textareaValue: value,
				hasValue: !!value,
			}));
		}
	}, [value]);

	const handleFocus = useCallback(() => {
		setState((prev) => ({ ...prev, isFocused: true }));
	}, []);

	const handleBlur = useCallback(
		(event: FocusEvent<HTMLTextAreaElement>) => {
			setState((prev) => ({
				...prev,
				isFocused: false,
				hasValue: !!prev.textareaValue,
			}));
			onBlur?.(event);
		},
		[onBlur]
	);

	const handleChange = useCallback(
		(event: ChangeEvent<HTMLTextAreaElement>) => {
			const newValue = event.target.value;
			setState((prev) => ({
				...prev,
				textareaValue: newValue,
				hasValue: !!newValue,
			}));
			onChange?.(event);
		},
		[onChange]
	);
```

Reply:
changed